### ISSUE TITLE
[Bug]: The Sponsored shortcuts option will re-enable after updating Fenix

### ISSUE THREAD

**Issue by LaurentiuApahideanSV** [mozilla-mobile/fenix#27092]
### Steps to reproduce

1. Install Fenix 104.2.0;
2. Go to Settings -> Homepage and disable or make sure Sponsored shortcuts are disabled;
3. Update Fenix to 105.1.0;
4. Go to Settings -> Homepage and observe the Sponsored shortcuts option and then check the homepage while connected to a VPN to the United States.

### Expected behaviour

After the update the Sponsored shortcuts should be disabled and sponsored tiles should not be displayed on the homepage.

### Actual behaviour

After the update the Sponsored shortcuts are re-enabled and sponsored tiles are displayed on the homepage.

### Devices used:
- Samsung Galaxy S22 Ultra (Android 12)
- Google Pixel 6 (Android 13)
- Samsung A32 (Android 12)

### Firefox release type

Firefox

### Firefox version

105.1.0

### Additional information

https://user-images.githubusercontent.com/50983519/191551660-af2334f4-5739-4a86-93aa-540fdefd9c78.mp4

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21624)

**Comment by Mugurell** [comment]
@LaurentiuApahideanSV 
I want to confirm that for the above tests in 104 there was no interaction with the "Sponsored shortcuts" toggle.
And that if the user is to toggle off "Sponsored shortcuts" then in 105 this action is remembered and "Sponsored shortcuts" is disabled.

**Comment by LaurentiuApahideanSV** [comment]
@Mugurell In the above tests there was no interaction with the "Sponsored shortcuts" toggle before updating Fenix.
I retested the issue by toggling the "Sponsored shortcuts" on and off before updating Fenix and in this case "Sponsored shortcuts" were not re-enabled after updating Fenix.

**Comment by Mugurell** [comment]
Then I'd say there is no bug.
The very important detail here is the user interaction - as only when the user actually toggles this setting (and any other in the app) the app will "remember" the user option. Otherwise with no user option the defaults will be applied and these may change over time, like it happened with Contile being enabled by default in 105 - https://github.com/mozilla-mobile/fenix/issues/26335.

To summarize the expected scenarios:
- If the user doesn't interact with the setting then:
    - in 104 the default is applied which is `disabled`
    (there was also an A/B experiment around this time that would change the default to `enabled` and this would apply if the user didn't interact with the setting)
    - in 105 the default is applied which is `enabled`.
- If the user does change this setting to `disabled` - the app will remember this and:
    - in 104 the user option will override the default and set Contile to `disabled`
    - in 105 the user option will override the default and set Contile to `disabled`.

**Comment by deusexcalamus** [comment]
This happened to me too, had sponsored shortcuts disabled, was less than pleased to find them re-enabled upon upgrade to 105.

**Comment by Mugurell** [comment]
> This happened to me too, had sponsored shortcuts disabled, was less than pleased to find them re-enabled upon upgrade to 105.

Did you specifically toggled them off in settings (maybe after first toggling them on) or they off by default?

**Comment by deusexcalamus** [comment]
> > This happened to me too, had sponsored shortcuts disabled, was less than pleased to find them re-enabled upon upgrade to 105.
> 
> Did you specifically toggled them off in settings (maybe after first toggling them on) or they off by default?

I turned them off after whenever they first showed up.

**Comment by Mugurell** [comment]
I tested again the 104 -> 105 update with 
 - simulating the feature getting enabled by an experiment in 104
 - me explicitly disabling "Sponsored shortcuts" in settings
 - updating to 105
 
 There were no sponsored shortcuts on the homescreen and the setting was still showing as disabled in settings.
 Given the importance of this scenario I'll add the qa-label again for one more check by QA also.

**Comment by Mugurell** [comment]
@deusexcalamus Can you share the exact scenario which showed "Sponsored shortcuts" enabled in settings after the update to 105?
Any preconditions?
Did you enabled and disabled the feature by yourself in 104?
Was it enabled by default in 104 and you remember disabling it before the update to 105?

**Comment by deusexcalamus** [comment]
> @deusexcalamus Can you share the exact scenario which showed "Sponsored shortcuts" enabled in settings after the update to 105? Any preconditions? Did you enabled and disabled the feature by yourself in 104? Was it enabled by default in 104 and you remember disabling it before the update to 105?

I imagine it was enabled by default whenever the feature first landed, and I noticed it, was disgusted by it, and turned it off. After I upgraded to 105 I noticed it was re-enabled and that there were sponsored tiles on my homepage. It's been so long since it was introduced, I can't really remember. I do know that it is something that I would turn off immediately if given the chance, though.

**Comment by SoftVision-LorandJanos** [comment]
- After I installed 104.2.0, I checked the setting for Sponsored shortcuts, and it was disabled - which was by default.
I then updated to 105.1.0, where Sponsored shortcuts were enabled - also by default.

- I then reinstalled 104.2.0 after removing the app entirely, where it was disabled by default, turned it ON manually, and updated to 105.1.0 again where it stayed ON.

- For the final scenario I had again a clean install of 104.2.0 where the feature is OFF by default, I then turned the feature ON, then OFF, after which I updated to the 105.1.0 where the feature remained OFF.

The results for all the scenarios above are as expected, so as @Mugurell commented above, I don't think either that this would be a bug, I think it can be closed.
Device used: Google Pixel 4 (Android 12).
